### PR TITLE
Run upgrades participant api tests on Admin API

### PR DESCRIPTION
**Pull request by dylant-da**
On top of Tudor's changes here: https://github.com/DACH-NY/canton/pull/17194
Basically ports tests on Ledger API (from here https://github.com/digital-asset/daml/pull/18431) to also run against Admin API

### CODE REVIEW

File: daml-lf/validation/BUILD.bazel [hunk 1]
```diff
@@ -51,6 +52,28 @@ da_scala_library(
...
+da_scala_library(
+    name = "validation-upgrades-test-lib",
+    srcs =
+        glob(
+            ["src/test/**/UpgradesSpecUtil.scala"],
+        ) +
+        #  we depend on sources to avoid pushing a canton artifact to maven
+        [
+            "//canton:community_admin-api_proto_scala",
+        ],
```
Comment (samuel-williams-da) [inline daml-lf/validation/BUILD.bazel:64]: Consider depending directly on a built canton artifact, like community_base, over recompiling the source code. Daml-script only does this as it is distributed, your tests are not

File: daml-script/test/BUILD.bazel [hunk 2]
```diff
@@ -253,7 +253,7 @@ da_scala_library(
         srcs = [":upgrades/coin-v3/Coin.daml"],
         project_name = "coin",
         target = lf_version,
-        upgrades = "//daml-script/test:coin-v2-{}.dar".format(lf_version),
+        # upgrades = "//daml-script/test:coin-v2-{}.dar".format(lf_version),
```
Comment (samuel-williams-da) [inline daml-script/test/BUILD.bazel:256]: can/should these be deleted? Also why this change - seems unrelated

**Review by samuel-williams-da — APPROVED** [review]
Noice!

File: daml-lf/validation/BUILD.bazel (hunk 1 above)
Comment (dylant-da) [reply daml-lf/validation/BUILD.bazel:64]: For some reason I couldn't get it to build without copying this in as a source - got a bunch of odd errors.

Comment (dylant-da) [reply daml-lf/validation/BUILD.bazel:64]: I tried to get this to work - will ping in morning, but for now going to merge so as not to stop the train

File: daml-script/test/BUILD.bazel (hunk 2 above)
Comment (samuel-williams-da) [reply daml-script/test/BUILD.bazel:256]: Ah right, makes sense
